app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -7,6 +7,26 @@ def run
           results = SerializableArray.new(results)
           SearchTaskDefinition.create_output(results, output)
         end
+
+        def self.dereference_output(storage, path)
```

Comment:
@tstrachota, Transformations are already a necessity. Right now it's pretty limited and implemented implicitly by square brackets (the transform function is a filter transform - you can only choose a property). After introducing custom output objects, it will be impossible to maintain same transformation code in framework, since the input will become unknown.
I am not a big fan of thread pool either and I don't think it's the ultimate solution. Still we have to address at least the problem with passing (transformed) output as a parameter to plan phase of another task. It is possible to finish the plan phase of the envelope task before the plan phase of dependent task starts. The difficulty here is that planning of tasks that rely on other task's output can be done only after the run phase of the dependency task has  returned with a result.

The example was a stupid one, I just couldn't think of a good one with a condition. Will this example be better:

``` Ruby
  host_creation = plan_action(CreateHost, host_attrs)
  if host_creation.output[:environment] == 'production'
    plan_action(AddServerToSplunkMonitoring, host_creation.output[:ip])
  end
```